src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -855,7 +867,7 @@ def save_as_table(
         # in save as table. So we rename ${number} with COL{number}.
         hidden_column_pattern = r"\"\$(\d+)\""
         column_definition_with_hidden_columns = attribute_to_schema_string(
-            child_attributes
+            child_attributes or []
```

Comment:
i think I would prefer the column definition to be None also, which indicate clear that we do not have column definition

Reply:
in fact, if the column definition is [], it should be an error state for general case, right?